### PR TITLE
Add calibration to provenance tracking

### PR DESCRIPTION
**Pull request by Bultako**
This PR extends the provenance products files and graph produced to previous calibration steps (as shown in example prov graph [calibration_to_dl2_01808_prov.pdf](https://github.com/cta-observatory/lstosa/files/7838240/calibration_to_dl2_01808_prov.pdf)). It adds to the provenance tracking the calibration processes that create pedestal, time and charge calibration files used in already tracked process `ro_to_dl1`, as well as the check plots files produced. Consequently prov products files are now renamed:

- calibration_to_dl1_runID_prov.pdf
- calibration_to_dl2_runID_prov.pdf

For the moment, the calibration processes tracked are those involved in the **next day processing**, those using external lstchain onsite calibration scripts, only input and output files relative to those scripts are tracked skipping the rest of parameters hard coded in those scripts and outside the scope of lstosa. In the future, provenance tracking for **massive reprocessing** will be added. See https://github.com/cta-observatory/lstosa/issues/54#issuecomment-993253782 for more details.

It also improves and refactor existing provenance tracking code.

### CODE REVIEW

**Comment by morcuended** [comment]
Thanks, @Bultako. I'll have a look.

File: osa/scripts/calibration_pipeline.py [hunk 1]
```diff
@@ -90,14 +91,19 @@ def calibration_sequence(drs4_pedestal_run_id: str, pedcal_run_id: str) -> int:
...
+    drs4_pedestal_run_id: str, pedcal_run_id: str, history_file: Path
+) -> int:
     """
     Create a DRS4 pedestal file for baseline correction.
 
     Parameters
     ----------
     drs4_pedestal_run_id : str
         String with run number of the pedestal run
+    pedcal_run_id : str
```
Comment (morcuended) [inline osa/scripts/calibration_pipeline.py:105]: why adding this argument here? Is it needed for provenance?

Comment (Bultako) [reply osa/scripts/calibration_pipeline.py:105]: Yes.
It is needed to identify link each both two calibration tasks to a single calibration workflow identified with a label of two numbers, namely `drs4_pedestal_run_id-pedcal_run_id`.

**Comment by morcuended** [comment]
Looks good to me. I'm going to test it in the container.

**Comment by Bultako** [comment]
I think we have to define a boundary on which provenance info `lstosa` should keep. I'm tempted to say that all info defined outside `lstosa` should not be kept, so external scripts used like those from `lstchain` would appear as blackboxes with inputs and outputs files that may be also connected to other tasks in `lstosa`. The reason why to do this is to simplify maintenance. We could also decide to add the detailed provenance info happening in these external boxes, in that case there is now a lot of info missing, and any modif in those `lstchain` scripts used will need to modify the code for provenance tracking in `lstosa`.

**Comment by morcuended** [comment]
> I think we have to define a boundary on which provenance info `lstosa` should keep. I'm tempted to say that all info defined outside `lstosa` should not be kept, so external scripts used like those from `lstchain` would appear as blackboxes with inputs and outputs files that may be also connected to other tasks in `lstosa`. The reason why to do this is to simplify maintenance. We could also decide to add the detailed provenance info happening in these external boxes, in that case there is now a lot of info missing, and any modif in those `lstchain` scripts used will need to modify the code for provenance tracking in `lstosa`.

Agreed. At least for now, we should at least keep track of which calibration files are used in r0_to_dl1 (which we have been doing so far). The information on the production of calibration files produced by osa is limited to the drs4 pedestal and calibration files and corresponding run numbers. We have no straight access to the time file (although in lstosa we look for this file in the same way for using it in r0 to dl1) nor the factor systematic file.

File: osa/provenance/config/definition.yaml [hunk 2]
```diff
@@ -296,14 +368,26 @@ entities:
         description: "Pedestal file in FITS format on the disk"
         type: File
         contentType: application/fits
```
Comment (morcuended) [inline osa/provenance/config/definition.yaml:370]: Pedestal file is now produced in h5 format.

**Review by morcuended — APPROVED** [review]

**Pull request merged**